src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -358,41 +358,67 @@ internal sealed class RSession : IRSession, IRCallbacks {
-                }
+                    if (!IsRemote && startupInfo.WorkingDirectory != null) {
+                        await evaluation.SetWorkingDirectoryAsync(startupInfo.WorkingDirectory);
+                    } else {
+                        await evaluation.SetDefaultWorkingDirectoryAsync();
+                    }
 
-                var callback = _callback;
-                if (callback != null) {
-                    await evaluation.SetVsGraphicsDeviceAsync();
-
-                    string mirrorUrl = callback.CranUrlFromName(startupInfo.CranMirrorName);
-                    await evaluation.SetVsCranSelectionAsync(mirrorUrl);
-
-                    await evaluation.SetCodePageAsync(startupInfo.CodePage);
-                    await evaluation.SetROptionsAsync();
-                    await evaluation.OverrideFunctionAsync("setwd", "base");
-                    await evaluation.SetFunctionRedirectionAsync();
-                    await evaluation.OptionsSetWidthAsync(startupInfo.TerminalWidth);
-
-                    try {
-                        // Only enable autosave for this session after querying the user about any existing file.
-                        // This way, if they happen to disconnect while still querying, we don't save the new empty
-                        // session and overwrite the old file.
-                        bool deleteExisting = await evaluation.QueryReloadAutosaveAsync();
-                        await evaluation.EnableAutosaveAsync(deleteExisting);
-                    } catch (REvaluationException) {
+                    var callback = _callback;
+                    if (callback != null) {
+                        await evaluation.SetVsGraphicsDeviceAsync();
+
+                        string mirrorUrl = callback.CranUrlFromName(startupInfo.CranMirrorName);
+
+                        try {
+                            await evaluation.SetVsCranSelectionAsync(mirrorUrl);
+                        } catch (REvaluationException ex) {
+                            await WriteErrorAsync(Resources.Error_SessionInitializationMirror, mirrorUrl, ex.Message);
+                        }
+
+                        try {
+                            await evaluation.SetCodePageAsync(startupInfo.CodePage);
+                        } catch (REvaluationException ex) {
+                            await WriteErrorAsync(Resources.Error_SessionInitializationCodePage, startupInfo.CodePage, ex.Message);
+                        }
+
+                        try {
+                            await evaluation.SetROptionsAsync();
+                        } catch (REvaluationException ex) {
+                            await WriteErrorAsync(Resources.Error_SessionInitializationOptions, ex.Message);
+                        }
+
+                        await evaluation.OverrideFunctionAsync("setwd", "base");
+                        await evaluation.SetFunctionRedirectionAsync();
+
+                        try {
```

Comment:
Perhaps make some `ProtectedEval` or `IgnorableEval` extension with optional print of the message? I think there are a few other places that perform similar things.